plugins/org.locationtech.udig.catalog/src/org/locationtech/udig/catalog/ID.java

```diff
@@ -468,7 +468,7 @@ public boolean isWFS() {
      * @return true if ID refers to a database (ie is a jdbc url)
      */
     public boolean isJDBC() {
-        return id.startsWith("jdbc"); //$NON-NLS-1$
+        return id.startsWith("jdbc") || (url != null && url.getProtocol().contains("jdbc")) ; //$NON-NLS-1$ //$NON-NLS-2$
```

Comment:
ID provides various constructors which do not necessary guarantee that id value is always derived from the URL. Moreover the GeoResource implementation provided for  different databases do not directly use the jdbc URL passed to a java JDBC connection. To give an example with such IDs produced fot Postgres and Mysql:
- Postgres --> example id value **jdbc.postgis**://postgres@pc06543:5432/db_name#testbed
- Mysql --> example id value **mysql.jdbc**://user:psswd@localhost:3306/db_name#test_1 (see also attached screenshot)
In the mysql case it is evident that id does not start with _jdbc_
![image](https://cloud.githubusercontent.com/assets/26147785/26538270/44fe631a-444e-11e7-93c0-ca9bd879011c.png)